Makefile

```diff
@@ -19,6 +19,42 @@ SWAGGER_ENDPOINTS_SPEC = apps/aeutils/src/endpoints.erl
 
 PACKAGE_SPEC_WIN32 ?= ../ci/appveyor/package.cfg
 
+AE_DEB_VERSION=`cat VERSION`
+AE_DEB_DCH_REL_NOTE= \
+"Release notes are available in /usr/share/doc/aeternity-node/docs/release-notes/RELEASENOTES-`cat VERSION`.md"
+
+AE_DEB_EMAIL="info@aeternity.com"
+AE_DEB_NAME="Aeternity Team"
+
+
+# Packages from master MUST be pre-releases. Git master version
+# usually is always higher then the last stable release. However
+# packages with newer stable version MUST always have higher version
+# than master in Debian/Ubuntu packaging context. The only way to
+# achieve this is when master packages are a pre-release (
+# pkg-name_version~unique_higher_number ).
+
+# Additionally the same (as in name) package from master for the same
+# unreleased/not stable version (i.e. builds) MUST always have higher
+# version (i.e. pre-release). Otherwise package managers and repository management
+# software complain.
+
+# Test source for master branch. Needs special treatment for Debian
+# packages.
+ifeq (, $(shell which git))
+  ifneq (,$(shell cat .git/HEAD | grep master))
```

Comment:
This is extremely fragile. What are use cases?

In local copy I never checkout master branch. I am usually in detached HEAD mode (`origin/master`, or other) or branch with other name. Also, if checked out on a branch, the branch name may sometimes contain master (e.g. `master--check?`). Is this only for CI?

Would it be possible to rather reverse the logic, by setting `AE_DEB_VERSION` to the content of the VERSION file only if a special var (`AE_DEB_STABLE`?) is passed by the caller, and always use this pre-release (complex) version otherwise?

Reply:
I like the idea. I'm also not OK "gitting" here. What if `AE_DEB_VERSION` defaults to contents of `VERSION` file and can be overwritten by the caller ?

Also I don't like the variable name as it means "debian version" to me. `AE_DEB_PKG_VERSION` sounds better and more specific

Reply:
I agree it is not the cleanest solution. It is targeted mainly at CI. The idea is that it should not be related to specific technology (CircleCI; variables etc).  One should be able to run `make prod-deb-package` and build a package on ~any shell. Imagine users checking out the code and trying to generate a package for themselves. 

The reverse logic will solve the master issue, but will require passing variables. I'm not a big fan of that. This will run in CI when a tag is created, and I am not aware how we will check for it and pass this variable then. I'll think about it tomorrow.

Reply:
My proposition is to rework this to check if the version in the VERSION file has a related tag. A stable package should be created. If it doesn't, a pre-release one (the code is from a branch). 

Example:

```
 git tag | grep `cat VERSION`
```
```
ifneq (, $(shell  git tag | grep `cat VERSION`))
   AE_DEB_VERSION=`cat VERSION`"-1-git~"`date +%Y%m%d%H%M%S`
endif
```
Potentially a git hash could also be added, if it does not break Debian versioning scheme and pre-releases incremental state.

The standard situation is to produce a package for a stable version. The pre-releases and builds from master (or whatever) are a deviation. In my understanding the approach  should be to create a stable release and add work-arounds on top for the deviation. I am not in favor of adding work-around (passing variables) to produce a stable version package. 

If we are two to one for passing a variable for a stable build, I'll do it, but I do not think it is the right approach.

Reply:
I don't see what's the problem and misunderstanding here.

`AE_DEB_PKG_VERSION ` should default to `cat VERSION`, without any gits mits dates or whatever.

External scripts must handle more exotic versioning models if they needs to by passing the new version:
e.g.
`make deb-prod-package AE_DEB_PKG_VERSION=3.0.0-bla-bla-bla`

Reply:
And more concrete the CI master job will run:

`
make deb-prod-package AE_DEB_PKG_VERSION=$(cat VERSION)-1-git-master~$(date +%Y%m%d%H%M%S)
`
or whatever the versioning scheme is

Reply:
> Also I don't like the variable name as it means "debian version" to me. `AE_DEB_PKG_VERSION` sounds better and more specific

Addressed in bce08c4

Reply:
> I don't see what's the problem and misunderstanding here.
> 
> `AE_DEB_PKG_VERSION ` should default to `cat VERSION`, without any gits mits dates or whatever.
> 
> External scripts must handle more exotic versioning models if they needs to by passing the new version:

Fine. I'll do it the wrong way. This leaves room for error and the possibility to generate stable package version (no pre-release) out of master (or whatever branch). My desire was for it to be self-sustained.

Reply:
Addressed in 93b75a5